### DISCUSSION TITLE
Model n_features change from the number present in data?

### DISCUSSION THREAD

**Discussion by Benzidrine** [pycaret/pycaret#2123] [Q&A] [answered]
If I do the following:

`setup(data = df, 
             target = "label",
             numeric_imputation = 'mean',
             silent = True, normalize = True, session_id=42)`

Then do something simple like:

`extratrees = create_model('et')`

`X = df.iloc[:,1:].values`

`extratrees.predict(X)`

I will get the following error:
`Number of features of the model must match the input. Model n_features is 1483 and input n_features is 784`

How are you meant to create models with Pycaret? Is this a bug?

**Comment by ngupta23** [comment, marked as answer]
What happens when you use `predict_model(extratrees, data=X)` instead?

**Reply by Benzidrine** [reply]
That's the approach I needed. Thank you very much for your answer.
